pkg/kubectl/cmd/rollout/rollout_status.go

```diff
@@ -0,0 +1,146 @@
+	Filenames []string
+	Recursive bool
+}
+
+const (
+	status_long    = `watch the status of current rollout.`
+	status_example = `# Watch the rollout status of a deployment
+$ kubectl rollout status deployment/abc`
+)
+
+func NewCmdRolloutStatus(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &StatusOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "status (TYPE NAME | TYPE/NAME) [flags]",
+		Short:   "Watch rollout status",
+		Long:    status_long,
+		Example: status_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunStatus(f, cmd, out, args, options))
+		},
+	}
+
+	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	return cmd
+}
+
+func RunStatus(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, args []string, options *StatusOptions) error {
+	if len(args) == 0 && len(options.Filenames) == 0 {
+		return cmdutil.UsageError(cmd, "Required resource not specified.")
+	}
+
+	mapper, typer := f.Object(false)
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, options.Recursive, options.Filenames...).
+		ResourceTypeOrNameArgs(true, args...).
+		SingleResourceType().
+		Latest().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("rollout status is only supported on individual resources and resource collections - %d resources were found", len(infos))
+	}
+	info := infos[0]
+	mapping := info.ResourceMapping()
+	if kind := mapping.GroupVersionKind.GroupKind(); kind != extensions.Kind("Deployment") {
+		return fmt.Errorf("rollout status isn't supported for %q", kind)
+	}
+
+	obj, err := r.Object()
+	if err != nil {
+		return err
+	}
+	rv, err := mapping.MetadataAccessor.ResourceVersion(obj)
+	if err != nil {
+		return err
+	}
+
+	// watch for changes to the deployment
+	w, err := r.Watch(rv)
+	if err != nil {
+		return err
+	}
+
+	statusViewer, err := f.StatusViewer(mapping)
+	if err != nil {
+		return err
+	}
+
+	// check if deployment's has finished the rollout
+	status, done, err := statusViewer.Status(cmdNamespace, info.Name)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "%s", status)
+	if done {
+		return nil
+	}
+
+	// if the rollout isn't done yet, keep watching deployment status
+	kubectl.WatchLoop(w, func(e watch.Event) error {
+		// print deployment's status
+		status, done, err := statusViewer.Status(cmdNamespace, info.Name)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "%s", status)
+		// Quit waiting if the rollout is done
+		if done {
+			w.Stop()
```

Comment:
Because we're in a watch loop, without this line users will be stuck in this loop even when the rollout is done. I'd like to stop watching after it's done. 

I have another question, if we're doing `w.Stop()` when the rollout is done, and in WatchLoop `w.Stop()` is called either when error happens or the interrupt signal is received. Do we still need the `defer w.Stop()`?